sphinxlint/__main__.py

```diff
@@ -118,13 +123,35 @@ def _check_file(todo):
     return check_file(*todo)
 
 
-def print_results(results):
-    """Print results (or a message if nothing is to be printed)."""
+def sort_errors(results, sorted_by):
+    """Flattens and potentially sorts errors based on user prefernces"""
+    if not sorted_by:
+        for results in results:
+            yield from results
+        return
+    errors = list(error for errors in results for error in errors)
+    # sorting is stable in python, so we can sort in reverse order to get the
+    # ordering specified by the user
+    for field in reversed(sorted_by):
+        if field == 'error_type':
+            errors.sort(key=lambda error: error.checker_name)
+        elif field == "filename":
+            errors.sort(key=lambda error: error.filename)
+        elif field == "line":
+            errors.sort(key=lambda error: error.line_no)
+        else:
+            raise ValueError(f"Unsupported sorting field {field}")
+    yield from errors
+
+def print_errors(errors, sorted_by):
+    """Print errors (or a message if nothing is to be printed)."""
     qty = 0
-    for result in results:
-        for line in result:
-            print(line)
-            qty += 1
+    for error in errors:
+        if sorted_by and sorted_by[0] == "error_type":
+            print(f"{error.checker_name}: {error.filename}:{error.line_no}: {error.msg}")
+        else:
+            print(f"{error.filename}:{error.line_no}: {error.msg} ({error.checker_name})")
```

Comment:
I think I understand why you did this, but I'm not 100% convinced it's a good idea: it can be surprising that a sort option change the output. If the output is parsed it means adding a sort option breaks the parsing.

In the other hand, if the output is parsed, there's clearly no need to sort it... so nobody should "hit" this "bug"...

If we have a single way to format things, you could implement it in a `__str__` in the dataclass.

Reply:
I appreciate you understanding why I did this, but I also understand the points your're raising, so I'll revert this then to the previous uniform printing, which I'll do by implementing a `__str__` function on the dataclass.